src/main/kotlin/org/pkl/intellij/psi/PklModuleImpl.kt

```diff
@@ -254,7 +254,13 @@ class PklModuleImpl(viewProvider: FileViewProvider) :
     }
 
   override val isInPklProject: Boolean
-    get() = pklProjectDir != null
+    get() = pklProjectDir != null && !isExcluded(project, virtualFile)
+
+  private fun isExcluded(project: Project, file: VirtualFile): Boolean {
+    val excludes =
+      project.modules.flatMap { ModuleRootManager.getInstance(it).excludeRoots.toList() }
+    return excludes.any { VfsUtilCore.isAncestor(it, file, true) }
+  }
```

Comment:
I think this is too expensive in its current form :(

We either need to find a way to make this cheap (maybe cache it, with correct cache invalidation), or let's just move it back to `PklSyncNotificationProvider`.

Reply:
This is executed once each time you open or switch to a new file. I can't see this being too expensive unless you have thousands of excluded folders.

Reply:
Ah, right, we currently only call this inside `PklSyncProjectNotificationProvider`. I had thought that we use this method on every resolve.